bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,839 @@
+
+    /**
+     * This handles the scheduling of an hourly forecast poll, to be applied with the given delay.
+     * When run, if requests the run-time of the next one is calculated and scheduled.
+     */
+    MetOfficeDelayedExecutor dailyForecastJob = new MetOfficeDelayedExecutor(scheduler);
+
+    /**
+     * This handles the scheduling of an hourly forecast poll, to be applied with the given delay.
+     * When run, if requests the run-time of the next one is calculated and scheduled.
+     */
+    MetOfficeDelayedExecutor hourlyForecastJob = new MetOfficeDelayedExecutor(scheduler);
+
+    public MetOfficeDataHubSiteApiHandler(Thing thing, IHttpClientProvider httpClientProvider,
+            @Reference LocationProvider locationProvider, @Reference TranslationProvider translationProvider,
+            @Reference LocaleProvider localeProvider, @Reference TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.locationProvider = locationProvider;
+        this.httpClient = httpClientProvider.getHttpClient();
+        this.translationProvider = translationProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(getClass());
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void dispose() {
+        dailyForecastJob.cancelScheduledTask(true);
+        hourlyForecastJob.cancelScheduledTask(true);
+        cancelDataRequiredCheck();
+        cancelScheduleDailyDataPoll(true);
+        if (initTask != null) {
+            initTask.cancel(true);
+            initTask = null;
+        }
+        super.dispose();
+    }
+
+    public String getLocalizedText(String key, @Nullable Object @Nullable... arguments) {
+        String result = translationProvider.getText(bundle, key, key, localeProvider.getLocale(), arguments);
+        return Objects.nonNull(result) ? result : key;
+    }
+
+    protected State getQuantityTypeState(@Nullable Number value, Unit<?> unit) {
+        return (value == null) ? UnDefType.UNDEF : new QuantityType<>(value, unit);
+    }
+
+    protected State getDecimalTypeState(@Nullable Number value) {
+        return (value == null) ? UnDefType.UNDEF : new DecimalType(value);
+    }
+
+    private void checkDataRequired() {
+        final List<@Nullable String> activeGroups = getThing().getChannels().stream().filter(x -> isLinked(x.getUID()))
+                .map(x -> x.getUID().getGroupId()).distinct().toList();
+
+        if (activeGroups.stream().anyMatch(g -> g != null && g.startsWith(GROUP_PREFIX_DAILY_FORECAST))) {
+            boolean repollRequired = !requiresDailyData;
+            if (!requiresDailyData) {
+                logger.trace("Daily data poll potentially required if older than required poll window");
+            }
+
+            requiresDailyData = true;
+            if (repollRequired) {
+                reconfigureDailyPolling();
+            }
+            logger.trace("Daily data poll required");
```

Comment:
Simplify
```suggestion
            boolean repollRequired = !requiresDailyData;
            requiresDailyData = true;
            if (repollRequired) {
                logger.trace("Daily data poll potentially required if older than required poll window");
                reconfigureDailyPolling();
            }
            logger.trace("Daily data poll required");
```

Reply:
Polling has been refactored in this last pass, this is hopefully cleaner now 🤞